tests/math/test_functions.py

```diff
@@ -258,6 +258,27 @@ def test_allclose(t1, t2):
 class TestAllCloseSparse:
     """Test that the sparse-matrix specialized allclose functions works well"""
 
+    @pytest.mark.parametrize("v", [0, 1, 2.0, 3.0j, 1e-9])
+    def test_sparse_scalar(self, v):
+        """Test comparing a scalar to a sparse matrix"""
+        dense = v
+
+        sparse = sp.sparse.csr_matrix([v] * 10)
+
+        assert fn.allclose(dense, sparse)
+        assert fn.allclose(sparse, dense)
+
+        # Shift one element of the sparse matrix
+        sparse_wrong = sp.sparse.csr_matrix([v + 0.1] + [v] * 9)
+        assert not fn.allclose(dense, sparse_wrong)
+        assert not fn.allclose(sparse_wrong, dense)
+
+        # Empty one element of the sparse matrix
+        v_nonzero = not np.isclose(v, 0)
+        sparse_wrong = sp.sparse.csr_matrix([0 if v_nonzero else 1] + [v] * 9)

```

Comment:
Just to make sure I understand correctly: For `v_nonzero=True`, this triggers the scenario where the sparse object is missing an entry (i.e., it is actually "sparse" mathematically speaking) and the `sparse_mat.nnz = np.prod(sparse_mat.shape)` test fails?

Reply:
Yes. By comparing an array to a scalar we mean that every element of the array should be close to the scalar. In our implementation of such comparison we 'abused' the sparsity of scipy sparse matrices so sometimes (i.e. before this PR) we might lose attention to those zeros and decide that a sparse is close to a scalar as long as it's non-zero values are close no matter how many zeros far away from the scalar.